File: Ecat.Client/core/common/mapStrings.ts

export class MpGender {
    static unk = 'Unknown';
    static male = 'Male';
    static female = 'Female';
    static other = 'Other';
}

export class MpAffiliation {
    static usaf = 'Air Force';
    static usa = 'Army';
    static uscg = 'Coast Guard';
    static usn = 'Navy';
    static usmc = 'Marines';
    static fn = 'Foreign National';
    static none = 'Unaffiliated';
    static unk = 'Unknown';
}

export class MpComponent {
    static active = 'Active Duty';
    static reserve = 'Reserves';
    static guard = 'National Guard';
    static none = 'Unaffiliated';
    static unk = 'Unknown';
}

export class MpPaygrade {
    static e1 = 'E1';
    static e2 = 'E2';
    static e3 = 'E3';
    static e4 = 'E4';
    static e5 = 'E5';
    static e6 = 'E6';
    static e7 = 'E7';
    static e8 = 'E8';
    static e9 = 'E9';
    static civ = 'Civilian';
    static fn = 'Foreign National';
    static unk = 'Unknown';
}

export class MpCommentFlag {
    static pos = "Positive";
    static neg = "Negative";
    static neut = "Neutral";
    static appr = "Appropriate";
    static inappr = "Inappropriate";
}

export class MpCommentType {
    static signed = "Signed";
    static anon = "Anonymous";
}

export class MpSpItemResponse {
    static iea = 'IEA';
    static ieu = 'IEU';
    static nd = 'ND';
    static eu = 'EU';
    static ea = 'EA';
    static heu = 'HEU';
    static hea = 'HEA';
}

export class MpApiResource {
    static user = 'User';
    static mock = 'Mock';
    static student = 'Student';
    static faculty = 'Faculty';
    static sa = 'SysAdmin';
    static courseAdmin = 'CourseAdmin';
    static designer = 'Designer';
}

export class MpAlertState {
    static info = 'info';
    static danger = 'danger';
    static success = 'success';
    static warning = 'warning';
    static default = 'inverse';
}

export class MpEdLevel {
    static cla = 'Chief Leadership Academy';
    static sncoa = 'Senior NCO Academy';
    static ncoa = 'NCO Academy';
    static als = 'Airman Leadership';
    static epmeic = 'EPMEIC';
    static none = 'None';
}

export class MpInstituteRole {
    static hqAdmin = 'ECAT_01_HQ';
    static designer = 'ECAT_02_Designer';
    static courseAdmin = 'ECAT_03_CourseAdmin';
    static faculty = 'ECAT_04_Faculty';
    static student = 'ECAT_05_Student';
    static external = 'ECAT_06_External';
}

export class MpAnimation {
    static fadeIn = 'fadeIn';
    static fadeInLeft = 'fadeInLeft';
    static fadeInRight = 'fadeInRight';
    static fadeInUp = 'fadeInUp';
    static fadeInDown = 'fadeInDown';
    static bounceIn = 'bounceIn';
    static bounceInLeft = 'bounceInLeft';
    static bounceInRight = 'bounceInRight';
    static bounceInUp = 'bounceInUp';
    static rotateInDownRight = 'rotateInDownRight';
    static rotateIn = 'rotateIn';
    static flipInX = 'flipInX';
    static flipInY = 'flipInY';
    static fadeOut = 'fadeOut';
    static fadeOutLeft = 'fadeOutLeft';
    static fadeOutRight = 'fadeOutRight';
    static fadeOutUp = 'fadeOutUp';
    static fadeOutDown = 'fadeOutDown';
    static bounceOut = 'bounceOut';
    static bounceOutLeft = 'bounceOutLeft';
    static bounceOutRight = 'bounceOutRight';
    static bounceOutUp = 'bounceOutUp';
    static rotateOutUpRight = 'rotateOutUpRight';
    static rotateOut = 'rotateOut';
    static flipOutX = 'flipOutX';
    static flipOutY = 'flipOutY';
}

export class MpEntityType {
    static unk = 'Unknown';
    static person = 'Person';
    static security = 'Security';
    static loginTk = 'LoginToken';
    static facProfile = 'ProfileFaculty';
    static hqStaffProfile = 'ProfileStaff';
    static studProfile = 'ProfileStudent';
    static externalProfile = 'Profileexternal';
    static course = 'Course';
    static academy = 'Academy';
    static workGroup = 'WorkGroup';
    static crseStudInGrp = 'CrseStudentInGroup';
    static studCrseMember = 'StudentInCourse';
    static facCrseMember = 'FacultyInCourse';
    static spInstr = 'SpInstrument';
    static kcInstr = 'KcInstrument';
    static ecInstr = 'Instrument';
    static spInventory = 'SpInventory';
    static kcResult = 'KcResult';
    static spResponse = 'SpResponse';
    static spComment = 'StudSpComment';
    static spCommentFlag = 'StudSpCommentFlag';
    static spStrat = 'StratResponse';
    static spResult = 'SpResult';
    static facSpResponse = 'FacSpResponse';
    static facSpComment = 'FacSpComment';
    static facSpCommentFlag = 'FacSpCommentFlag';
    static facStratResponse = 'FacStratResponse';
}

export class MpSweetAlertType {
    static success = 'success';
    static warn = 'warning';
    static err = 'error';
    static info = 'info';
    static input = 'input';
}

export class MpSpStatus {

    static und = 'Udr';
    static open = 'Open';
    static underReview = 'Under Review';
    static published = 'Published';
    static arch = 'ArchiveE';
}

export class MpSpResult {
    static ie = 'Ineffective';
    static bae ='Below Avg Effectiveness';
    static e = 'Effective';
    static aae = 'Above Avg Effectiveness';
    static he = 'High Effective';
}

export class MpBehaviorResult {
    static hea = 'Always: Highly Effective';
    static heu = 'Usually: Highly Effective';
    static ea = 'Always: Effective';
    static eu = 'Usually: Effective';
    static ieu = 'Usually: Ineffective';
    static iea = 'Always: Ineffective';
}

File: Ecat.Shared.Core/Utility/MapStrings.cs
namespace Ecat.Shared.Core.Utility
{
    public static class MpStrings
    {
    }

    //public static class MpTableNames
    //{
    //    public const string SpAssessResp = "SpStudResponse",
    //        SpAssessResult = "SpStudResult",
    //        GrpMember = "GroupMember",
    //        Person = "Person";
    //}


    public static class MpInstituteRoleId
    {
        public const string HqAdmin = "ECAT_01_HQ",
            Designer = "ECAT_02_Designer",
            CourseAdmin = "ECAT_03_CourseAdmin",
            Faculty = "ECAT_04_Faculty",
            Student = "ECAT_05_Student",
            External = "ECAT_06_External",
            RefOnly = "ECAT_07_RefOnly",
            Undefined = "Undefined";
    }

    public static class MpInstituteRoleName
    {
        public const string HqAdmin = "BCEE: ECAT 01 HQ",
            Designer = "BCEE: ECAT 02 Designer",
            CourseAdmin = "BCEE: ECAT 03 Course Admin",
            Facilitator = "BCEE: ECAT 04 Faculty",
            Student = "BCEE: ECAT 05 Student",
            External = "BCEE: ECAT 06 External",
            RefOnly = "BCEE: ECAT 07 Reference Only";
    }


    public static class MpTransform
    {
        public static RoleMap InstituteRoleToEnum(string instituteRole)
        {
            switch (instituteRole)
            {
                case MpInstituteRoleId.HqAdmin:
                    return RoleMap.SysAdmin;
                case MpInstituteRoleId.Designer:
                    return RoleMap.Designer;
                case MpInstituteRoleId.CourseAdmin:
                    return RoleMap.Faculty;
                case MpInstituteRoleId.Faculty:
                    return RoleMap.Faculty;
                case MpInstituteRoleId.Student:
                    return RoleMap.Student;
                case MpInstituteRoleId.External:
                    return RoleMap.External;
                case MpInstituteRoleId.RefOnly:
                    return RoleMap.RefOnly;
                default:
                    return RoleMap.Unknown;
            }
        }

        public static string RoleNameToId(string roleName)
        {
            switch (roleName)
            {
                case MpInstituteRoleName.HqAdmin:
                    return MpInstituteRoleId.HqAdmin;
                case MpInstituteRoleName.CourseAdmin:
                    return MpInstituteRoleId.CourseAdmin;
                case MpInstituteRoleName.Designer:
                    return MpInstituteRoleId.Designer;
                case MpInstituteRoleName.Facilitator:
                    return MpInstituteRoleId.Faculty;
                case MpInstituteRoleName.Student:
                    return MpInstituteRoleId.Student;
                case MpInstituteRoleName.External:
                    return MpInstituteRoleId.External;
                case MpInstituteRoleName.RefOnly:
                    return MpInstituteRoleName.RefOnly;

                default:
                    return MpInstituteRoleId.Undefined;
            }
        }
    }

    public static class MpGroupType
    {
        public const string Wg1 = "BC1",
              Wg2 = "BC2",
              Wg3 = "BC3",
              Wg4 = "BC4",
              None = "None";
    }

    public static class MpCommentType
    {
        public const string Signed = "Signed",
            Anon = "Anonymous";
    }

    public static class MpCommentFlag
    {
        public const string Pos = "Postiive",
            Neg = "Negative",
            Neut = "Netural";
    }

    public static class MpCourseRole
    {
        public const string Student = "Student",
            Faculty = "Faculty",
            CrseAdmin = "Course Admin",
            SysAdmin = "System Admin";
    }

    public static class MpELevel
    {
        public const string Cla = "Chief Leadership Academy",
              Sncoa = "Senior NCO Academy",
              Ncoa = "NCO Academy",
              Als = "Airman Leadership",
              Epmeic = "EPMEIC",
              None = "None";
    }

    public static class MpAssessResult
    {
        public const string He = "Highly Effective",
            Aae = "Above Avg Effective",
            Ae = "Average Effective",
            Bae = "Below Avg Effective",
            Ie = "Ineffective";
    }

    public static class MpAffiliation
    {
        public const string Usaf = "Air Force",
            Usa = "Army",
            Uscg = "Coast Guard",
            Usn = "Navy",
            Usmc = "Marines",
            Fn = "Foreign National",
            None = "Unaffiliated",
            Unk = "Unknown";
    }

    public static class MpGender
    {
        public const string Unk = "Unknown",
            Other = "Other",
            Male = "Male",
            Female = "Female";
    }

    public static class MpComponent
    {
        public const string Active = "Active Duty",
               Reserve = "Reserves",
               Guard = "National Guard",
               None = "Unaffiliated",
               Unk = "Unknown";
    }

    public static class MpPaygrade
    {
        public const string E1 = "E1",
             E2 = "E2",
             E3 = "E3",
             E4 = "E4",
             E5 = "E5",
             E6 = "E6",
             E7 = "E7",
             E8 = "E8",
             E9 = "E9",
             Civ = "Civilian",
             Fn = "Foreign National",
             Unk = "Unknown";
    }

    public static class MpSpStatus
    {
        public const string Und = "Udr",
            Open = "Open",
            UnderReview = "Under Review",
            Published = "Published",
            Arch = "ArchiveE";
    }

    public static class MpSpItemResponse
    {
        public const string
            Iea = "IEA",
            Ieu = "IEU",
            Nd = "ND",
            Eu = "EU",
            Ea = "EA",
            Heu = "HEU",
            Hea = "HEA";
    }

    public static class MpToken
    {
        public static string[] TokenMethods => new[] {
            "Context.WS:login",
            "Context.WS:loginTool",
            "Context.WS:emulateUser",
            "Context.WS:extendSessionLife",
            "Context.WS:getMemberships",
            "Context.WS:getMyMemberships",
            "Context.WS:initialize",
            "Course.WS:changeCourseBatchUid",
            "Course.WS:changeCourseCategoryBatchUid",
            "Course.WS:changeCourseDataSourceId",
            "Course.WS:changeOrgBatchUid",
            "Course.WS:changeOrgCategoryBatchUid",
            "Course.WS:changeOrgDataSourceId",
            "Course.WS:createCourse",
            "Course.WS:createOrg",
            "Course.WS:getAvailableGroupTools",
            "Course.WS:getCartridge",
            "Course.WS:getCategories",
            "Course.WS:getClassifications",
            "Course.WS:getCourse",
            "Course.WS:getCourseCategoryMembership",
            "Course.WS:getGroup",
            "Course.WS:getOrg",
            "Course.WS:getOrgCategoryMembership",
            "Course.WS:getServerVersion",
            "Course.WS:getStaffInfo",
            "Course.WS:initializeCourse",
            "Course.WS:saveCartridge",
            "Course.WS:saveCourse",
            "Course.WS:saveCourseCategory",
            "Course.WS:saveGroup",
            "Course.WS:saveOrgCategory",
            "Course.WS:saveOrgCategoryMembership",
            "Course.WS:saveStaffInfo",
            "Course.WS:setCourseBannerImage",
            "Course.WS:updateCourse",
            "Course.WS:updateOrg",
            "Announcement.WS:getCourseAnnouncements",
            "Announcement.WS:getOrgAnnouncements",
            "Announcement.WS:getSystemAnnouncements",
            "Announcement.WS:initializeAnnouncementWS",
            "Announcement.WS:createCourseAnnouncements",
            "Announcement.WS:createOrgAnnouncements",
            "Announcement.WS:createSystemAnnouncements",
            "Announcement.WS:updateCourseAnnouncements",
            "Announcement.WS:updateOrgAnnouncements",
            "Announcement.WS:updateSystemAnnouncements",
            "Announcement.WS:getServerVersion",
            "Gradebook.WS:getAttempts",
            "Gradebook.WS:getGradebookColumns",
            "Gradebook.WS:getGradebookTypes",
            "Gradebook.WS:getGrades",
            "Gradebook.WS:getGradingSchemas",
            "Gradebook.WS:getRequiredEntitlements",
            "Gradebook.WS:getServerVersion",
            "Gradebook.WS:initializeGradebookWS",
            "Gradebook.WS:saveAttempts",
            "Gradebook.WS:saveColumns",
            "Gradebook.WS:saveGradebookTypes",
            "Gradebook.WS:saveGrades",
            "Gradebook.WS:saveGradingSchemas",
            "CourseMembership.WS:getCourseMembership",
            "CourseMembership.WS:getCourseRoles",
            "CourseMembership.WS:getGroupMembership",
            "CourseMembership.WS:getServerVersion",
            "CourseMembership.WS:initializeCourseMembershipWS",
            "Content.WS:addContentFile",
            "Content.WS:updateContentFileLinkName",
            "Content.WS:getContentFiles",
            "Content.WS:getFilteredContent",
            "Content.WS:getReviewStatusByCourseId",
            "Content.WS:getFilteredCourseStatus",
            "Content.WS:getLinksByReferrerType",
            "Content.WS:getLinksByReferredToType",
            "Content.WS:getTOCsByCourseId",
            "Content.WS:initializeContentWS",
            "Content.WS:initializeVersion2ContentWS",
            "Content.WS:getRequiredEntitlements",
            "Content.WS:saveContent",
            "Content.WS:saveCourseTOC",
            "Content.WS:saveContentsReviewed",
            "NotificationDistributorOperations.WS:getServerVersion",
            "NotificationDistributorOperations.WS:initializeNotificationDistributorOperationsWS",
            "NotificationDistributorOperations.WS:setDistributorAvailabilityForUser",
            "NotificationDistributorOperations.WS:registerDistributorResults",
            "NotificationDistributorOperations.WS:",
            "User.WS:getServerVersion",
            "User.WS:initializeUserWS",
            "User.WS:saveUser",
            "User.WS:saveObserverAssociation",
            "User.WS:saveAddressBookEntry",
            "User.WS:getUser",
            "User.WS:getAddressBookEntry",
            "User.WS:getObservee",
            "User.WS:changeUserBatchUid",
            "User.WS:changeUserDataSourceId",
            "User.WS:getSystemRoles",
            "User.WS:getInstitutionRoles",
            "User.WS:getUserInstitutionRoles",
            "Calendar.WS:initializeCalendarWS",
            "Calendar.WS:createPersonalCalendarItem",
            "Calendar.WS:createCourseCalendarItem",
            "Calendar.WS:createInstitutionCalendarItem",
            "Calendar.WS:updatePersonalCalendarItem",
            "Calendar.WS:updateCourseCalendarItem",
            "Calendar.WS:updateInstitutionCalendarItem",
            "Calendar.WS:savePersonalCalendarItem",
            "Calendar.WS:saveCourseCalendarItem",
            "Calendar.WS:saveInstitutionCalendarItem",
            "Calendar.WS:getCalendarItem",
            "Calendar.WS:canUpdatePersonalCalendarItem",
            "Calendar.WS:canUpdateCourseCalendarItem",
            "Calendar.WS:canUpdateInstitutionCalendarItem",
            "Util.WS:initializeUtilWS",
            "Util.WS:getRequiredEntitlements",
            "Util.WS:checkEntitlement",
            "Util.WS:saveSetting",
            "Util.WS:loadSetting",
            "Util.WS:deleteSetting",
            "Util.WS:getDataSources"
        };
    }

}
